controllers/cluster_controller.go

```diff
@@ -162,6 +154,28 @@ func (r *ClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr e
 	return r.reconcile(ctx, cluster)
 }
 
+func patchCluster(patchHelper *patch.Helper, ctx context.Context, cluster *clusterv1.Cluster, options ...patch.Option) error {
```

Comment:
The Patch operation is becoming really verbose so I moved it to a separate func.
This func is designed to be used also for issuing patches in the middle of the reconcile loop, if necessary